url7f43a4752f_shalynnho_pokemonfactory_tgz-pJ8-Networking_ServerJ8/src/DeviceGraphicsDisplay/KitRobotGraphicsDisplay.java

```diff
@@ -0,0 +1,577 @@
+		jobIsDone = false;
+		initialJob = true;
+		this.moveToInitialPosition = initialCommand;
+		this.moveToFinalPosition = finalCommand;
+	}
+
+	// begin paths
+	public void InspectionToLocation1() {
+		setCommands(Command.moveToInspectionStand, Command.moveToLocation1);
+		moveToInitialOrFinal();
+	}
+
+	public void InspectionToLocation2() {
+		setCommands(Command.moveToInspectionStand, Command.moveToLocation2);
+		moveToInitialOrFinal();
+	}
+
+	public void InspectionToGoodConveyor() {
+		setCommands(Command.moveToInspectionStand, Command.moveToGoodConveyor);
+		moveToInitialOrFinal();
+	}
+
+	public void ConveyorToLocation1() {
+		setCommands(Command.moveToConveyor, Command.moveToLocation1);
+		moveToInitialOrFinal();
+	}
+
+	public void ConveyorToLocation2() {
+		setCommands(Command.moveToConveyor, Command.moveToLocation2);
+		moveToInitialOrFinal();
+	}
+
+	public void Location1ToInspectionStand() {
+		setCommands(Command.moveToLocation1, Command.moveToInspectionStand);
+		moveToInitialOrFinal();
+	}
+
+	public void Location2ToInspectionStand() {
+		setCommands(Command.moveToLocation2, Command.moveToInspectionStand);
+		moveToInitialOrFinal();
+	}
+
+	public void InspectionStandToGoodConveyor() {
+		setCommands(Command.moveToInspectionStand, Command.moveToGoodConveyor);
+		moveToInitialOrFinal();
+	}
+
+	public void Location1ToLocation2() {
+		setCommands(Command.moveToLocation1, Command.moveToLocation2);
+		moveToInitialOrFinal();
+	}
+
+	// end paths
+
+	/*
+	 * sets the rotation configurations based on the commands
+	 */
+	public void moveToInitialOrFinal() {
+
+		animateKitRobot = true;
+		if (initialJob) {
+			moveToPosition = moveToInitialPosition;
+			invisibleCloud = false;
+		} else if (finalJob) {
+			moveToPosition = moveToFinalPosition;
+			invisibleCloud = true;
+		} else if (returnJob) {
+			moveToPosition = Command.moveToConveyor;
+		}
+
+		if (position.equals(Position.conveyorPosition)) {
+			if (moveToPosition.equals(Command.moveToInspectionStand)) {
+				setVectorConfigurations(inspectionLocation.getX()
+						- kitEnterLocation.getX(), inspectionLocation.getY()
+						- kitEnterLocation.getY(), Position.inspectionPosition);
+			} else if (moveToPosition.equals(Command.moveToLocation1)) {
+				setVectorConfigurations(
+						location1.getX() - kitEnterLocation.getX(),
+						location1.getY() - kitEnterLocation.getY(),
+						Position.location1Position);
+			} else if (moveToPosition.equals(Command.moveToLocation2)) {
+				setVectorConfigurations(
+						location2.getX() - kitEnterLocation.getX(),
+						location2.getY() - kitEnterLocation.getY(),
+						Position.location2Position);
+			} else if (moveToPosition.equals(Command.moveToGoodConveyor)) {
+				setVectorConfigurations(kitLeaveLocation.getX()
+						- kitEnterLocation.getX(), kitLeaveLocation.getY()
+						- kitEnterLocation.getY(),
+						Position.goodConveyorPosition);
+			} else {
+				setVectorConfigurations(0, 0, Position.conveyorPosition);
+			}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 199:220, 222:243, 245:266, 268:289, 290:312.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.